src/cargo/sources/git/utils.rs

```diff
@@ -355,12 +365,33 @@ impl<'a> GitCheckout<'a> {
                     "Skipping",
                     format!(
                         "git submodule `{}` due to update strategy in .gitmodules",
-                        url
+                        child_url_str
                     ),
                 )?;
                 return Ok(());
             }
 
+            // See the `git submodule add` command on the documentation:
+            // https://git-scm.com/docs/git-submodule
```

Comment:
To reduce the need for people jumping to external documentation, could you call out what from that you find relevant.

For example

> git only assumes a URL is a relative file path if it starts with `./` or `../`
>
> See [`git submodule add` documentation](https://git-scm.com/docs/git-submodule)